nix-libs/nodeLib/buildNodePackage.nix

```diff
@@ -593,10 +593,13 @@ let
         env = buildNodePackage (args // {includeDevDependencies = true;});
 
         # An 'overrideNodePackage' attribute, which will call
-        # `buildNodePackage` with the given arguments overridden.
+        # `buildNodePackage` with new arguments produced by the given
+        # arg-override function. The function consumes the original
+        # argument set.
+        #
         # We don't use the name `override` because this will get stomped on
         # if the derivation is the result of a `callPackage` application.
-        overrideNodePackage = newArgs: buildNodePackage (args // newArgs);
+        overrideNodePackage = newArgsFun: buildNodePackage (args // (newArgsFun args));
```

Comment:
I like having this behavior, but since this would change the original behavior of the function, it would break existing code. We can use duck typing to accomplish both at once, with something like 

```
if builtins.isFunction newArgs 
then buildNodePackage (args // (newArgs args))
else buildNodePackage (args // newArgs);
```